File: 1346.check-if-n-and-its-double-exist.py
#
# @lc app=leetcode id=1346 lang=python
#
# [1346] Check If N and Its Double Exist
#

# @lc code=start
class Solution(object):
    def checkIfExist(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """

        for i in range(len(arr)):
            for j in range(len(arr)):
                if i != j and arr[i] == 2 * arr[j]:
                    return True

        return False

# @lc code=end

File: 1346.check-if-n-and-its-double-exist.js
/*
 * @lc app=leetcode id=1346 lang=javascript
 *
 * [1346] Check If N and Its Double Exist
 */

// @lc code=start
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var checkIfExist = function(arr) {
    const map = {};

    for (let i = 0; i < arr.length; i++) {
        const num = arr[i];
        if (map[num * 2] || map[num / 2]) {
            return true;
        }
        map[num] = true;
    }
    return false;
};
// @lc code=end
